web-console/src/views/datasource-view/datasource-view.tsx

```diff
@@ -224,7 +225,7 @@ interface RetentionDialogOpenOn {
 
 interface CompactionDialogOpenOn {
   readonly datasource: string;
-  readonly compactionConfig: CompactionConfig;
+  readonly compactionConfig?: CompactionConfig;
```

Comment:
Why this change and the ` | undefined` below?

Reply:
Because a CompactionDialog can be open on a datasource without a compaction config (like when one is being created for the first time). This was already being set to `undefined` in some places, this change just makes the issue more explicit.